### PR TITLE
[enahancement] Added app loader page 

### LINKED ISSUE #4388: [enhancement] Should manage errors when a user tries to access an app it doesn't have access
### Is your proposal related to a problem?

Now, it Opens the app editor when a user tries to access an app it doesn't have access to.

### Describe the solution you'd like

Redirect to another page (Dashboard or show 403 error page)

### If the feature is approved, would you be willing to submit a PR?
Yes

### PR DESCRIPTION
**Pull request by shah21**
Resolves #4388

### CODE REVIEW

**Comment by shubh22** [comment]
Task linked: [CU-2x0tkdb Manage Forbidden/Unauthorised app errors](https://app.clickup.com/t/2x0tkdb)

File: frontend/src/AppLoader/AppLoader.jsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+      })
+      .catch((error) => {
+        setErrorDetails(error);
+        setLoading(false);
+      });
+  };
+
+  const handleError = () => {
+    if (errorDetails?.data) {
+      const statusCode = errorDetails.data?.statusCode;
```
Comment (gsmithun4) [inline frontend/src/AppLoader/AppLoader.jsx:34]: Shall we change the get apps API and get the application's organization id in error case?
and we can call the switch organization API

**Review by gsmithun4 — CHANGES_REQUESTED** [review]

**Comment by gsmithun4** [comment]
deployed https://tj-app-loader.herokuapp.com/

File: frontend/src/Editor/Viewer.jsx [hunk 2]
```diff
@@ -152,19 +157,69 @@ class ViewerComponent extends React.Component {
...
+        return (window.location.href = `login/${orgId}?redirectTo=${path}`);
+      }
+    );
+  };
+
+  handleError = (errorDetails, appId, versionId) => {
+    if (errorDetails?.data) {
+      const statusCode = errorDetails.data?.statusCode;
+      if (statusCode === 403) {
+        const errorObj = JSON.parse(errorDetails.data?.message);
```
Comment (gsmithun4) [inline frontend/src/Editor/Viewer.jsx:211]: we should consider the scenario when `errorDetails.data?.message` empty or not valid json. try catch can be added and on error redirected to home page, same for editor loader

**Review by gsmithun4 — CHANGES_REQUESTED** [review]

File: frontend/src/_helpers/utils.js [hunk 3]
```diff
@@ -530,3 +530,15 @@ export const hightlightMentionedUserInComment = (comment) => {
...
+  let parsed;
+
+  try {
+    parsed = JSON.parse(json);
+  } catch (e) {
+    console.log('JSON parse error');
+  }
+
+  return parsed;
+}
```
Comment (gsmithun4) [inline frontend/src/_helpers/utils.js:544]:
```suggestion
export function safelyParseJSON(json) {
  try {
    return JSON.parse(json);
  } catch (e) {
    console.log('JSON parse error');
  }
  return;
}
```

**Review by gsmithun4 — APPROVED** [review]

**Pull request merged**